src/internet_identity/src/main.rs

```diff
@@ -175,6 +175,9 @@ struct Challenge {
 thread_local! {
     static STATE: State = State::default();
     static ASSETS: RefCell<HashMap<&'static str, (Vec<HeaderField>, &'static [u8])>> = RefCell::new(HashMap::default());
+    /// This build's version info, provided on startup through arguments.
```

Comment:
Is there a specific reason why you chose to allow setting the version at startup / upgrade? Because this allows to mismatch the `wasm` file and the version. Could we not bake the version into the `wasm` during the build? I think, in all contexts where II gets built we do have access to the git hash.

Reply:
In general it's not advised, because the build should not depend on being run in Git. In particular, we do not have access to `.git` in docker.